### DISCUSSION TITLE
Can I add standard deviation to the output report?

### DISCUSSION THREAD

**Discussion by TomerG2** [artilleryio/artillery#2305] [Artillery]
I was wondering if it's possible to add a standard deviation to the output of my run?
If not is it possible to add/implement such a metric?
(Any additional info would be most appriciated)

**Comment by hassy** [comment]
hi @TomerG2 👋 Artillery does not have support for reporting standard deviation right now. It's possible to sort-of do it yourself in an [`afterResponse` hook](https://www.artillery.io/docs/reference/engines/http#afterresponse-hooks) which have access to the response time for each request, but it will have significant limitations:

* The value won't be reported as an Artillery metric (you'll have to just `console.log` it)
* The calculation won't work for distributed tests on Lambda/Fargate

What's your use-case for std dev, is that something you need for analysis where percentiles aren't sufficient, or a requirement you have for reporting the results elsewhere?

**Reply by TomerG2** [reply]
Hi @hassy, thanks for the quick response!
I'm adding a jitter to an API in order to mitigate time based email enumeration attack,
I wanted to measure the STD to get an idea of how much jitter to add in order to sufficiently 
obscure the information with minimal hindrance to user experience
